Cubical/Foundations/Isomorphism.agda

```diff
@@ -0,0 +1,94 @@
+{-
+
+Theory about isomorphisms
+
+- Definitions of [section] and [retract]
+- Definition of isomorphisms ([Iso])
+- Any isomorphism is an equivalence ([isoToEquiv])
+
+-}
+{-# OPTIONS --cubical --safe #-}
+module Cubical.Foundations.Isomorphism where
+
+open import Agda.Primitive
+
+open import Cubical.Core.Everything
+open import Cubical.Foundations.HLevels
+
+-- Section and retract
+module _ {ℓ ℓ'} {A : Set ℓ} {B : Set ℓ'} where
+  section : (f : A → B) → (g : B → A) → Set ℓ'
+  section f g = ∀ b → f (g b) ≡ b
+
+  -- NB: `g` is the retraction!
+  retract : (f : A → B) → (g : B → A) → Set ℓ
+  retract f g = ∀ a → g (f a) ≡ a
+
+
+record isoStruct {ℓ ℓ'} {A : Set ℓ} {B : Set ℓ'} (f : A → B) : Set (ℓ ⊔ ℓ') where
```

Comment:
Why not put the f into the structure? 

The code would also look nicer if we had a constructor for this. For example `iso` or `_,_,_,_`.

Reply:
Yes, I can put one in.

Reply:
I guess I can call the whole structure "Iso" then.